test/export/StructureDefinitionExporter.test.ts

```diff
@@ -1595,22 +1595,21 @@ describe('StructureDefinitionExporter', () => {
     // Quantity is the first ancestor of Duration and Age
 
     // * value[x] only Duration or Age
-    // * value[x].comparator = #>=
+    // * value[x].comparator = #>
 
-    const profile = new Profile('QuantifiedObservation');
-    profile.parent = 'Observation';
+    const extension = new Extension('QuantifiedExtension');
 
     const onlyRule = new OnlyRule('value[x]');
     onlyRule.types = [{ type: 'Duration' }, { type: 'Age' }];
-    profile.rules.push(onlyRule);
+    extension.rules.push(onlyRule);
 
     const fixedValueRule = new FixedValueRule('value[x].comparator');
     fixedValueRule.fixedValue = new FshCode('>=');
-    profile.rules.push(fixedValueRule);
+    extension.rules.push(fixedValueRule);
 
-    exporter.exportStructDef(profile);
-    const sd = pkg.profiles[0];
-    const fixedElement = sd.findElement('Observation.value[x].comparator');
+    exporter.exportStructDef(extension);
+    const sd = pkg.extensions[0];
+    const fixedElement = sd.findElement('Extension.value[x].comparator');
```

Comment:
Just to make sure I understand: the type on `value[x]` would be `Quantity`. Would the type have profiles for `Age` and `Duration`, or would it have no profiles?

Reply:
In this case the type on `value[x]` does actually get set to `Duration` or `Age`. `Duration` and `Age` are specializations of `Quantity`, but they are treated just as if they were any other type, because `Quantity` is not an abstract type, so it cannot be replaced by either `Duration` or `Age`. So on the `Observation.value[x]`, you cannot constrain the type to `Duration`, because `Duration` is not one of the choices for that `value[x]` element. `Quantity` is, but we cannot replace `Quantity` with `Duration`. That is why I had to change this test to use `Extension`, since `Duration` and `Age` are both explicitly listed in the list of types the `Extension.value[x]` choice can take on.

Reply:
Ahh, okay. Thank you for clarifying!